core/src/queue.rs

```diff
@@ -220,64 +219,53 @@ impl Queue {
         }
     }
 
-    /// Pop a single transaction.
-    ///
-    /// Unlike [`Self::pop`], unsigned transactions are not recorded.
-    #[allow(
-        clippy::expect_used,
-        clippy::unwrap_in_result,
-        clippy::cognitive_complexity
-    )]
-    pub fn pop_without_seen(&self, wsv: &WorldStateView) -> Option<VersionedAcceptedTransaction> {
-        loop {
-            let hash = self.queue.pop()?;
-            let entry = match self.txs.entry(hash) {
-                Entry::Occupied(entry) => entry,
-                // As practice shows this code is not `unreachable!()`.
-                // When transactions are submitted quickly it can be reached.
-                Entry::Vacant(_) => continue,
-            };
-            if self.check_tx(entry.get(), wsv).is_err() {
-                entry.remove_entry();
-                continue;
-            }
-
-            if *entry
-                .get()
-                .check_signature_condition(wsv)
-                .expect("Checked in `check_tx` just above")
-            {
-                return Some(entry.get().clone());
-            }
-        }
-    }
-
-    /// Return the number of transactions in the queue
+    /// Return the number of transactions in the queue.
     pub fn tx_len(&self) -> usize {
         self.txs.len()
     }
 
     /// Gets transactions till they fill whole block or till the end of queue.
     ///
     /// BEWARE: Shouldn't be called in parallel with itself.
-    #[allow(clippy::missing_panics_doc, clippy::unwrap_in_result)]
     pub fn get_transactions_for_block(
```

Comment:
We can make this private, since it's used only in tests as far as I see, and swap name length with `put_transactions_for_block_into`.

Reply:
What do you mean by "swap name length"?

Reply:
Oh, sorry. I meant, we could rename them so that public one is shorter.

Reply:
Do you have any suggestion how we can make function name shorter? 

Because `put_transactions_for_block_into` and `get_transactions_for_block` a bit different in the first case we put transactions in the provided buffer and in the second case we return new vector.

Reply:
We could make `put_transactions_for_block_into` the new `get_transactions_for_block` and rename the one that provides the Vec to something like `collect_transactions_for_block`, or better yet, also move it to tests mod, since as far as I understand it's a convenience fn just for tests.

Reply:
Sorry, I'm bad with words today. Basically
`get_transactions_for_block` ->  `tests::collect_transactions_for_block` or something
`put_transactions_for_block_into` -> `get_transactions_for_block`

Reply:
good suggestion, i will do this